upstairs/src/volume.rs

```diff
@@ -672,7 +676,10 @@ pub enum VolumeConstructionRequest {
 }
 
 impl Volume {
-    pub fn construct(request: VolumeConstructionRequest) -> Result<Volume> {
+    pub fn construct(
+        request: VolumeConstructionRequest,
+        producer_registry: Arc<Mutex<Option<ProducerRegistry>>>,
```

Comment:
this should be `Option<Arc<Mutex<ProducerRegistry>>>` I think - it someone wants to pass None, it doesn't have to be wrapped in an arc and mutex.

Reply:
I tried to have it just an Option, but I'm passing the `ProducerRegistry` around on the propolis side and I was not able to get it to work without Arc Mutex.  @luqmana had the same comment on the propolis side PR, and I'm going to try from the other side to see if I can detangle this and if so, I'll come back here and change it as well.